### PR TITLE
Address problems

### PR DESCRIPTION
**Pull request by soring323**
- resolve transaction structure problem 
- update fee mechanism update

### CODE REVIEW

File: modules/apps/101-interchain-swap/keeper/msg_server.go [hunk 1]
```diff
@@ -196,11 +195,16 @@ func (k Keeper) OnCreatePoolReceived(ctx sdk.Context, msg *types.MsgCreatePoolRe
...
+	// Instantiate an interchain market maker with the default fee rate
+	amm, found := k.GetInterchainMarketMaker(ctx, pooId)
+	if !found {
+		amm = *types.NewInterchainMarketMaker(
+			&pool,
+			types.DefaultMaxFeeRate,
+		)
+		// save market maker
+		k.SetInterchainMarketMaker(ctx, amm)
+	}
```
Comment (liangping) [inline modules/apps/101-interchain-swap/keeper/msg_server.go:207]: AMM should be stateless.  can be initiated by parameters.  not sure why store this object.

File: modules/apps/101-interchain-swap/keeper/msg_server.go [hunk 2]
```diff
@@ -226,11 +230,16 @@ func (k Keeper) OnDepositReceived(ctx sdk.Context, msg *types.MsgDepositRequest)
...
-	)
+	// Instantiate an interchain market maker with the default fee rate
+	amm, found := k.GetInterchainMarketMaker(ctx, pool.PoolId)
+	if !found {
+		amm = *types.NewInterchainMarketMaker(
+			&pool,
+			types.DefaultMaxFeeRate,
+		)
+		// save market maker
+		k.SetInterchainMarketMaker(ctx, amm)
```
Comment (liangping) [inline modules/apps/101-interchain-swap/keeper/msg_server.go:241]: same here

**Review by liangping — COMMENTED** [review]
Structure looks more clear now.

File: modules/apps/101-interchain-swap/keeper/msg_server.go (hunk 2 above)
Comment (soring323) [reply modules/apps/101-interchain-swap/keeper/msg_server.go:241]: changed

**Pull request merged**